plugins/Wallet/js/index.js

```diff
@@ -0,0 +1,46 @@
+// index.js: main entrypoint for the Sia-UI wallet plugin.
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import createSagaMiddleware from 'redux-saga';
+import { createStore, applyMiddleware } from 'redux';
+import { Provider } from 'react-redux';
+import rootReducer from './js/reducers/index.js';
+import * as sagas from './js/sagas/wallet.js'
+import { getLockStatus, getBalance, getTransactions } from './js/actions/wallet.js';
+import WalletApp from './js/components/app.js';
+
+// Set up saga middleware system
+const sagaMiddleware = createSagaMiddleware();
+const store = createStore(
+	rootReducer,
+	applyMiddleware(sagaMiddleware)
+);
+sagaMiddleware.run(sagas.watchGetLockStatus);
+sagaMiddleware.run(sagas.watchUnlockWallet);
+sagaMiddleware.run(sagas.watchCreateNewWallet);
+sagaMiddleware.run(sagas.watchGetBalance);
+sagaMiddleware.run(sagas.watchGetTransactions);
+sagaMiddleware.run(sagas.watchGetNewReceiveAddress);
+sagaMiddleware.run(sagas.watchSendSiacoin);
```

Comment:
seems like the preferred way to handle this is a "root saga," as exemplified here:
https://github.com/yelouafi/redux-saga/blob/master/examples/real-world/index.js
https://github.com/yelouafi/redux-saga/blob/master/examples/real-world/sagas/index.js